src/routes/submission/index.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import SubmissionPage from './SubmissionPage';
+import { fetchSubmission } from '../../actions/submission';
+
+export default [
+  {
+    path: '/submission/:id',
+
+    async action({ params, store }) {
+      const submission = await store.dispatch(fetchSubmission(params.id));
```

Comment:
The action should handle the store dispatch part, I will make a commit to highlight it.